From beb01bfb68607bc03adbfd3f238419b238d20fb6 Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@gnome.org>
Date: Mon, 27 Sep 2010 17:15:04 +0000
Subject: gtk-engine: Only build it on gtk 2

As discussed on IRC, disable building the gtk engine on gtk 3.
Fixing it up would involve too much work, given that it's basically
unused.
---
diff --git a/Makefile.am b/Makefile.am
index 12d0d68..5338e5f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,10 @@
-SUBDIRS = . gdk-pixbuf-loader gtk-engine data tests tools doc
+SUBDIRS = . gdk-pixbuf-loader data tests tools doc
+
+if ENABLE_GTK_ENGINE
+if HAVE_GTK_2
+SUBDIRS += gtk-engine
+endif
+endif
 
 lib_LTLIBRARIES = librsvg-@RSVG_API_MAJOR_VERSION@.la
 
diff --git a/gtk-engine/Makefile.am b/gtk-engine/Makefile.am
index e8c21b4..09ee4aa 100644
--- a/gtk-engine/Makefile.am
+++ b/gtk-engine/Makefile.am
@@ -1,16 +1,12 @@
 SUBDIRS =
 DIST_SUBDIRS = examples
 
-if ENABLE_GTK_ENGINE
-
 SUBDIRS += examples
 
 gtk2_enginedir = $(libdir)/gtk-2.0/$(GTK2_BINARY_VERSION)/engines
-gtk3_enginedir = $(libdir)/gtk-3.0/$(GTK3_BINARY_VERSION)/engines
 
 themedir = $(datadir)/themes
 
-if HAVE_GTK_2
 gtk2_engine_LTLIBRARIES = libsvg.la
 
 libsvg_la_SOURCES = 		\
@@ -46,47 +42,3 @@ libsvg_la_LIBADD = \
 if PLATFORM_WIN32
 libsvg_la_LDFLAGS += -no-undefined
 endif
-
-endif # HAVE_GTK_2
-
-if HAVE_GTK_3
-gtk3_engine_LTLIBRARIES = libsvg-3.la
-
-libsvg_3_la_SOURCES = 		\
-	svg-draw.c		\
-	svg-main.c		\
-	svg-render.c		\
-	svg-rc-style.c		\
-	svg-rc-style.h		\
-	svg-style.h		\
-	svg.h
-
-libsvg_3_la_CPPFLAGS = \
-	-I$(top_srcdir)		\
-	-I$(top_builddir)	\
-	-DG_LOG_DOMAIN=\"rsvg-engine-3\" \
-	$(AM_CPPFLAGS)
-
-libsvg_3_la_CFLAGS = \
-	$(GTK3_CFLAGS)		\
-	$(LIBRSVG_CFLAGS)	\
-	$(LIBCROCO_CFLAGS) 	\
-	$(LIBGSF_CFLAGS) 	\
-	$(AM_CFLAGS)
-
-libsvg_3_la_LDFLAGS =  -avoid-version -module $(AM_LDFLAGS)
-
-libsvg_3_la_LIBADD = \
-	$(top_builddir)/librsvg-@RSVG_API_MAJOR_VERSION@.la \
-	$(GTK3_LIBS) 		\
-	$(LIBRSVG_LIBS) 	\
-	$(LIBCROCO_LIBS)	\
-	$(LIBGSF_LIBS)
-
-if PLATFORM_WIN32
-libsvg_3_la_LDFLAGS += -no-undefined
-endif
-
-endif # HAVE_GTK_3
-
-endif # ENABLE_GTK_ENGINE
--
cgit v0.8.3.1
